manager/integration/tests/test_migration.py

```diff
@@ -220,7 +223,70 @@ def test_migration_with_rebuilding_replica(clients, volume_name):  # NOQA
     7. Observe volume migrated to node 2 (single active engine).
     8. Validate initially written test data.
     """
-    return
+    # Step 1
+    client = get_random_client(clients)
+    current_host = common.get_self_host_id()
+    host1, host2 = get_hosts_for_migration_test(clients) # NOQA
+
+    # Step 2
+    volume = client.create_volume(name=volume_name, size=str(2 * Gi),
+                                  numberOfReplicas=3,
+                                  backingImage="",
+                                  accessMode="rwx", migratable=True)
+    volume = common.wait_for_volume_detached(client, volume_name)
+    volume.attach(hostId=current_host)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    volume_endpoint = get_volume_endpoint(volume)
+    write_volume_dev_random_mb_data(volume_endpoint,
+                                    1, 1500)
+    data = get_device_checksum(volume_endpoint)
+
+    # Step 3
+    replica0 = volume.replicas[0]
+    replica1 = volume.replicas[1]
+    replica2 = volume.replicas[2]
+
+    host_replica = get_host_replica(volume, host_id=current_host)
+    volume.replicaRemove(name=host_replica.name)
+    new_replica_found = False
+    for i in range(RETRY_COUNTS):
+        v = client.by_id_volume(volume_name)
+        for r in v.replicas:
+            if r.name != replica0.name and \
+                    r.name != replica1.name and \
+                    r.name != replica2.name:
+                new_replica_found = True
+                break
+        if new_replica_found:
+            break
+        time.sleep(RETRY_INTERVAL)
+
+    assert new_replica_found is True
+
+    # Step 4
+    volume.attach(hostId=host1)
+
+    # Step 5
+    volume = common.wait_for_volume_migration_ready(client, volume_name)
```

Comment:
Need to check the number of replicas from the first volume the same as the second migrated.

Reply:
I update code to check the new replica numbers for review, thank you.